owncloudApp/src/main/java/com/owncloud/android/ui/activity/ShareActivity.kt

```diff
@@ -0,0 +1,394 @@
+                    RemoteShare.FEDERATED_PERMISSIONS_FOR_FILE_AFTER_OC9
+            } else {
+                if (file.isFolder)
+                    RemoteShare.FEDERATED_PERMISSIONS_FOR_FOLDER_UP_TO_OC9
+                else
+                    RemoteShare.FEDERATED_PERMISSIONS_FOR_FILE_UP_TO_OC9
+            }
+        } else {
+            return if (file.isFolder)
+                RemoteShare.MAXIMUM_PERMISSIONS_FOR_FOLDER
+            else
+                RemoteShare.MAXIMUM_PERMISSIONS_FOR_FILE
+        }
+    }
+
+    override fun showSearchUsersAndGroups() {
+        val searchFragment = SearchShareesFragment.newInstance(file, account)
+        val ft = supportFragmentManager.beginTransaction()
+        ft.replace(R.id.share_fragment_container, searchFragment, TAG_SEARCH_FRAGMENT)
+        ft.addToBackStack(null)    // BACK button will recover the ShareFragment
+        ft.commit()
+    }
+
+    override fun showEditPrivateShare(share: OCShare) {
+        val ft = supportFragmentManager.beginTransaction()
+        val prev = supportFragmentManager.findFragmentByTag(TAG_EDIT_SHARE_FRAGMENT)
+        if (prev != null) {
+            ft.remove(prev)    // BACK button will recover the previous fragment
+        }
+        ft.addToBackStack(null)
+
+        // Create and show the dialog.
+        val newFragment = EditShareFragment.newInstance(share, file, account)
+        newFragment.show(ft, TAG_EDIT_SHARE_FRAGMENT)
+
+    }
+
+    override// Call to Remove share operation
+    fun removeShare(share: OCShare) {
+        fileOperationsHelper.removeShare(share)
+    }
+
+    /**
+     * Get users and groups from the server to fill in the "share with" list
+     */
+    override fun refreshSharesFromServer() {
+        // Show loading
+        showLoadingDialog(R.string.common_loading)
+        // Get Users and Groups
+        mGetSharesForFileAsyncTask = GetSharesForFileAsyncTask(this)
+        val params = arrayOf(file, account, storageManager)
+        mGetSharesForFileAsyncTask!!.execute(*params)
+    }
+
+    override fun showAddPublicShare(defaultLinkName: String) {
+        // DialogFragment.show() will take care of adding the fragment
+        // in a transaction.  We also want to remove any currently showing
+        // dialog, so make our own transaction and take care of that here.
+        val ft = supportFragmentManager.beginTransaction()
+        val prev = supportFragmentManager.findFragmentByTag(TAG_PUBLIC_SHARE_DIALOG_FRAGMENT)
+        if (prev != null) {
+            ft.remove(prev)
+        }
+        ft.addToBackStack(null)
+
+        // Create and show the dialog
+        val newFragment = PublicShareDialogFragment.newInstanceToCreate(
+            file,
+            account,
+            defaultLinkName
+        )
+        newFragment.show(ft, TAG_PUBLIC_SHARE_DIALOG_FRAGMENT)
+    }
+
+    override fun showEditPublicShare(share: OCShare) {
+        val ft = supportFragmentManager.beginTransaction()
```

Comment:
I see this code repeated several times.

Why don't you create an extension to Activity to show fragments??

Reply:
👍